### PR TITLE
Implemented Checkboxes, splash screen

### PR DESCRIPTION
**Pull request by tinawen02**
In this request, I worked on fixing the bug in my checkboxes where my program would throw an “index out of bounds” error. This ended up being because I never instantiated my checkedList in BillItem. I also switched over from a List<Boolean> to a SparseBooleanArray at Nygel’s suggestion because this would allow me to populate a certain index of checkedList without having to initialize it prior. In addition to this, I also finished writing calculateSplitSettlement() in DividingItemsActivity, where I created a Map<String, Double>. In this map, the keys represented the names of people and the values represented the amounts each person owed. I also took the time today to create a Splash Screen, although the logo is currently too large, so that’s something I’ll need to fix in following diffs. In the next few days, I’m planning on figuring out a way to actually display the information from amounts owed into SplitSettlementActivity as well as coming up with an algorithm for splitting a singular item amongst several people.

I tested my changes by using several breakpoints while writing my code, specifically when writing the calculateSplitSettlement() method. I wasn’t too sure of the logic behind this method, but using breakpoints allowed me to dig deeper into what each variable was actually storing and I was eventually able to figure out the logic by stepping over each line.

https://user-images.githubusercontent.com/84735390/175432554-11114d8d-4b18-491b-9dd0-6c2138ca007a.mov

### CODE REVIEW

File: app/src/main/res/drawable/splash_background.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
+    <item
+        android:drawable="@drawable/bg_gradient"/>
+    <item>
+        <bitmap android:gravity="center"
+            android:src="@drawable/iou_logo" />
```
Comment (Kampijo) [inline app/src/main/res/drawable/splash_background.xml:7]: Looking at the splash screen, maybe we can use a png with transparent background haha

File: app/src/main/java/com/example/iou/bill/adapters/BillItemAdapter.java [hunk 2]
```diff
@@ -85,7 +87,12 @@ public void bind(BillItem bill) {
                 checkbox.setOnClickListener(new View.OnClickListener() {
                     @Override
                     public void onClick(View v) {
-                        bill.toggleCheckbox(finalI, checkbox.isChecked());
+
+                        // Set the UI to display the check or uncheck
+                        checkbox.setChecked(!bill.getChecks(finalI));
+
+                        // Update the BillItem bill
+                        bill.toggleCheckbox(finalI, !bill.getChecks(finalI));
```
Comment (Kampijo) [inline app/src/main/java/com/example/iou/bill/adapters/BillItemAdapter.java:95]: A little bit of a nit, but we can make consolidate some logic here

`boolean checked = !bill.getChecks(finalI);`
 `checkbox.setChecked(checked);`
 `bill.toggleCheckBox(finalI, checked);`

File: app/src/main/java/com/example/iou/bill/activities/DividingItemsActivity.java [hunk 3]
```diff
@@ -74,31 +76,63 @@ public void onClick(View v) {
         });
     }
 
-    //TODO: calculate totals for each person
-    // Create a dict: keys are names, values are amounts owed
-    // Get the values from the Bill Item
-    // For loop iterating through items to get each billItem
-
+    // Does the math behind splitting a bill
```
Comment (Kampijo) [inline app/src/main/java/com/example/iou/bill/activities/DividingItemsActivity.java:79]: This is a hefty algorithm. We'd normally want a clear separation between UI and business logic.

I'd recommend actually bringing this all out into a separate class (e.g. `SplitSettlementCalculator.java`)

File: app/src/main/java/com/example/iou/activities/SplashScreenActivity.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+            @Override
+            public void run() {
+                //This method will be executed once the timer is over
+                // Start your app main activity
+                Intent i = new Intent(SplashScreenActivity.this, LoginActivity.class);
+                startActivity(i);
+                // close this activity
+                finish();
+            }
+        }, 1000);
```
Comment (Kampijo) [inline app/src/main/java/com/example/iou/activities/SplashScreenActivity.java:27]: do you have any plans to use this activity to initialize any dependencies that could potentially take a long time?

Comment (tinawen02) [reply app/src/main/java/com/example/iou/activities/SplashScreenActivity.java:27]: Not at the moment - I don't think I have dependencies that could take a long time right now.

**Pull request merged**